src/main/java/cleo/Storage.java

```diff
@@ -0,0 +1,131 @@
+package cleo;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import cleo.task.Deadline;
+import cleo.task.Events;
+import cleo.task.Task;
+import cleo.task.TaskList;
+import cleo.task.ToDos;
+
+/**
+ * Provides utility methods for storing and retrieiving
+ * tasks stored onto user's device.
+ */
+public class Storage {
+    private String filePath;
+    private SimpleDateFormat formatter = new SimpleDateFormat("MMM dd yyyy hh:mm a");
+    /**
+     * Creates a new Storage object with the specified file path.
+     * Ensures that the data folder exists.
+     * @param filePath
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        ensureDataFolderExists();
+
+    }
+
+    /**
+     * Ensures that the data folder exists, creating it if it doesn't.
+     */
+    private void ensureDataFolderExists() {
+        File dataFolder = new File("./data");
+        if (!dataFolder.exists()) {
+            dataFolder.mkdirs(); // Create the folder and any necessary parent directories
+            System.out.println("Data folder not found. Creating './data' directory now!");
+        }
+    }
```

Comment:
I really like how you abstracted out this part of the logic to create a new file if ./data doesn't exist. good job!